apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -595,12 +595,18 @@ function handleCloseEditor( calypsoPort ) {
 // The close button is generally overridden using the <MainDashboardButton> slot API
 // which was introduced in Gutenberg 8.2. In older editors we still need to override
 // the click handler so that the link will open in the parent frame instead of the
-// iframe. If this happens to be a newer version of the editor the <MainDashboardButton>
-// in `handleCloseEditor()` will end up overriding these chanages.
+// iframe. We try not to add the click handler if <MainDashboardButton> has been used.
 function handleCloseInLegacyEditors( handleClose ) {
-	const legacySelector = '.edit-post-fullscreen-mode-close__toolbar a'; // support for Gutenberg plugin < v7.7
-	const selector = '.edit-post-header .edit-post-fullscreen-mode-close';
+	// Selects close buttons in Gutenberg plugin < v7.7
+	const legacySelector = '.edit-post-fullscreen-mode-close__toolbar a';
+
+	// Selects the close button in modern Gutenberg versions, unless it itself is a close button override
+	const wpcomCloseSelector = '.wpcom-block-editor__close-button';
+	const navSidebarCloseSelector = '.wpcom-block-editor-nav-sidebar-toggle-sidebar-button__button';
+	const selector = `.edit-post-header .edit-post-fullscreen-mode-close:not(${ wpcomCloseSelector }):not(${ navSidebarCloseSelector })`;
```

Comment:
Just trying to make sure I understand this chained selector — this appears to target `.edit-post-header .edit-post-fullscreen-mode-close` if it has _neither_ the `.wpcom-block-editor__close-button` nor the `.wpcom-block-editor-nav-sidebar-toggle-sidebar-button__button` class?

Reply:
Yep that's right. I would write the logic as:
- match `.edit-post-header .edit-post-fullscreen-mode-close` unless:
  - it also matches the `.wpcom-block-editor__close-button` selector; or
  - it also matches the `.wpcom-block-editor-nav-sidebar-toggle-sidebar-button__button` selector

When you concat CSS selectors without using a space it means it has to match _all_ of the selectors. So I guess another way of thinking about it (ignoring the descendent selector) would be: select the element if
- it matches `.edit-post-fullscreen-mode-close` _and_
- it matches `:not(.wpcom-block-editor__close-button)` _and_
- it matches `:not(.wpcom-block-editor-nav-sidebar-toggle-sidebar-button__button)`

---

CSS is wild 😅

Reply:
Nice, thanks for the explanation! 😀